src/transformers/models/gpt2/modeling_flax_gpt2.py

```diff
@@ -425,8 +425,9 @@ class FlaxGPT2BlockCollection(nn.Module):
     dtype: jnp.dtype = jnp.float32
 
     def setup(self):
+        block_cls = nn.remat(FlaxGPT2Block, concrete=True) if self.config.gradient_checkpointing else FlaxGPT2Block
```

Comment:
Let's actually not make it dependent on the config here. @sgugger made a good point a while back that it shouldn't have been in the config in the first place. Can we maybe just add an input parameter to the model class init that is then called from the training scripts?